core/build.gradle.kts

```diff
@@ -187,6 +192,40 @@ ide {
     generatedSource(javaCCTest, "test")
 }
 
+fun JavaCompile.configureAnnotationSet(sourceSet: SourceSet) {
+    source = sourceSet.java
+    classpath = sourceSet.compileClasspath
+    destinationDirectory.set(options.generatedSourceOutputDirectory)
+    options.compilerArgs.add("-proc:only")
+    org.gradle.api.plugins.internal.JvmPluginsHelper.configureAnnotationProcessorPath(sourceSet, sourceSet.java, options, project)
+}
+
+val annotationProcessorMain by tasks.registering(JavaCompile::class) {
+    dependsOn(javaCCMain)
+    configureAnnotationSet(sourceSets.main.get())
+    // only if we aren't running compileJava, since doing twice fails.
+    onlyIf { project.gradle.taskGraph.hasTask("compileJava") }
```

Comment:
by the way, `compileJava` is available from `sourceSet.getCompileTaskName("java")`, so `onlyIf` could be moved to `configureAnnotationSet`

Reply:
Nice, updated.